ctapipe/irf/binning.py

```diff
@@ -0,0 +1,176 @@
+    ).tag(config=True)
+
+    reco_energy_min = Float(
+        help="Minimum value for Reco Energy bins in TeV units",
+        default_value=0.006,
+    ).tag(config=True)
+
+    reco_energy_max = Float(
+        help="Maximum value for Reco Energy bins in TeV units",
+        default_value=190,
+    ).tag(config=True)
+
+    reco_energy_n_bins_per_decade = Float(
+        help="Number of edges per decade for Reco Energy bins",
+        default_value=5,
+    ).tag(config=True)
+
+    energy_migration_min = Float(
+        help="Minimum value of Energy Migration matrix",
+        default_value=0.2,
+    ).tag(config=True)
+
+    energy_migration_max = Float(
+        help="Maximum value of Energy Migration matrix",
+        default_value=5,
+    ).tag(config=True)
+
+    energy_migration_n_bins = Integer(
+        help="Number of bins in log scale for Energy Migration matrix",
+        default_value=31,
+    ).tag(config=True)
+
+    def true_energy_bins(self):
+        """
+        Creates bins per decade for true MC energy using pyirf function.
+        """
+        true_energy = create_bins_per_decade(
+            self.true_energy_min * u.TeV,
+            self.true_energy_max * u.TeV,
+            self.true_energy_n_bins_per_decade,
+        )
+        return true_energy
+
+    def reco_energy_bins(self):
+        """
+        Creates bins per decade for reconstructed MC energy using pyirf function.
+        """
+        reco_energy = create_bins_per_decade(
+            self.reco_energy_min * u.TeV,
+            self.reco_energy_max * u.TeV,
+            self.reco_energy_n_bins_per_decade,
+        )
+        return reco_energy
+
+    def energy_migration_bins(self):
+        """
+        Creates bins for energy migration.
+        """
+        energy_migration = np.geomspace(
+            self.energy_migration_min,
+            self.energy_migration_max,
+            self.energy_migration_n_bins,
+        )
+        return energy_migration
+
+
+class FovOffsetBinning(Component):
+    """
+    Collects information on generating energy and angular bins for
+    generating IRFs as per pyIRF requirements.
+    """
+
+    fov_offset_min = Float(
+        help="Minimum value for FoV Offset bins in degrees",
+        default_value=0.0,
+    ).tag(config=True)
+
+    fov_offset_max = Float(
+        help="Maximum value for FoV offset bins in degrees",
+        default_value=5.0,
+    ).tag(config=True)
+
+    fov_offset_n_edges = Integer(
+        help="Number of edges for FoV offset bins",
+        default_value=2,
+    ).tag(config=True)
+
+    def fov_offset_bins(self):
+        """
+        Creates bins for single/multiple FoV offset.
+        """
+        fov_offset = (
+            np.linspace(
+                self.fov_offset_min,
+                self.fov_offset_max,
+                self.fov_offset_n_edges,
+            )
+            * u.deg
+        )
+        return fov_offset
+
+
+class SourceOffsetBinning(Component):
```

Comment:
This is specific to the PSF format, so it should only be configured there

Reply:
Made a Psf class.